src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -183,12 +191,24 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
     }
 
     public class RankerWeight extends Weight {
+        private final List<Expression> expressions;
         private final List<Weight> weights;
 
         RankerWeight(List<Weight> weights) {
             super(RankerQuery.this);
             assert weights instanceof RandomAccess;
             this.weights = weights;
+
+            // TODO: Can remove null check once derived features are implemented for stored queries
+            // Compile expressions for the derived features
+            if(features.derivedFeatures() != null) {
+                this.expressions = new ArrayList<>(features.derivedFeatures().size());
+                for (DerivedFeature df : features.derivedFeatures()) {
+                    expressions.add((Expression) Scripting.compile(df.expression()));
```

Comment:
I think we really need to compile and cache the expression, I think it'll be just fine to add a `private final Expression expression` to `StoredDerivedFeature`.